hpy/tools/autogen/ctx.py

```diff
@@ -40,6 +41,9 @@ def declare_func(self, func):
         # turn the function declaration into a function POINTER declaration
         newnode = deepcopy(func.node)
         newnode.type = c_ast.PtrDecl(type=newnode.type, quals=[])
+        if conf.RETURN_CONSTANT.get(func.node.name):
+            make_void(newnode)
```

Comment:
Please extract this to a helper function as the code repeats itself (in more than one variation).

Reply:
I assume you mean the `if` condition? Extracted to `get_return_constant`. Introducing some new `make_void_if_returns_constant(func, func_node)` seemed to me as not much more concise/self-explanatory than the plain `if`.

Reply:
You can call it `maybe_make_void` and it would be shorter.

Reply:
good suggestion, done